packages/react-core/src/components/Timestamp/Timestamp.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Timestamp/timestamp';
+import { css } from '@patternfly/react-styles';
+import { Tooltip, TooltipProps } from '../Tooltip';
+
+export enum TimestampFormats {
+  full = 'full',
+  long = 'long',
+  medium = 'medium',
+  short = 'short'
+}
+
+export interface TimestampProps extends React.HTMLProps<HTMLSpanElement> {
+  /** Custom content rendered inside the timestamp, such as a relative time. Passing this prop
+   * in will override the default content and timeZoneSuffix prop.
+   */
+  children?: React.ReactNode;
+  /** Class to add to the outer span. */
+  className?: string;
+  /** Applies custom formatting to the displayed date and/or time when custom content is not
+   * passed in. Passing this prop in will override the dateFormat and timeFormat props.
+   */
+  customFormat?: { [key: string]: string };
+  /** A date object that is used to render default content inside the timestamp. */
+  date?: Date;
+  /** Determines the format of the displayed date in the timestamp and UTC tooltip when custom
+   * content is not passed in. Examples:
+   * "Full" => Tuesday, August 9, 2022;
+   * "Long" => August 9, 2022;
+   * "Medium" => Aug 9, 2022;
+   * "Short" => 8/9/22
+   */
+  dateFormat?: 'full' | 'long' | 'medium' | 'short';
+  /** An ISO 8601 formatted date string to apply to the inner time element's datetime attribute.
+   * This can either be a UTC datetime string, or a local datetime string.
+   *
+   * For more information about the time element's datetime attribute:
+   * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/time#valid_datetime_values
+   */
+  datetime: string;
+  /** Flag indicating whether the timestamp has a tooltip to display the UTC time.
+   * To render a tooltip with custom content, pass in tooltipProps instead.
+   */
+  hasUTCTooltip?: boolean;
+  /** Flag for displaying the time in a 12 hour format (true) or a 24 hour format (false),
```

Comment:
The default values are show in the prop table.  I also don't think it is necessary to mention it is only used for custom content.  I think that is implied.   Maybe we can shorten this to:
"Flag for displaying the time in a 12 hour format. If this prop is not passed in, the hour format will be based on the locale prop's value."
Ee could also add a blurb somewhere explaining in implications of using children prop.

Reply:
For the "when custom content is not passed in" text, that was more to convey that this prop will only really work when displaying the default content. I could reword it a bit to either say "...when not using the children prop", or "...when the default content is displayed." (along with other areas where I mention "when custom content is not passed in"). What do you think?

Reply:
I feel like most of the props do not apply to custom content so it seems like instead of mentioning it for every prop, we could write a blurb somewhere,  maybe even in the description of the children prop.